recipe/meta.yaml

```
@@ -632,6 +632,20 @@ outputs:
         # not actually missing, but installed into SP_DIR, see tests
         - '*/arrow_python.dll'                # [win]
         - '*/arrow_python_flight.dll'         # [win]
+        # pyarrow-core builds with the capabilities but we do not ship them
+        # to provide the smaller core functionality.
+        - '*/libarrow_acero.so.*'        # [linux]
+        - '*/arrow_acero.dll'            # [win]
+        - '*/libarrow_dataset.so.*'      # [linux]
+        - '*/arrow_dataset.dll'          # [win]
+        - '*/libarrow_substrait.so.*'    # [linux]
+        - '*/arrow_substrait.dll'        # [win]
+        - '*/libarrow_flight.so.*'       # [linux]
+        - '*/arrow_flight.dll'           # [win]
+        - '*/libparquet.so.*'            # [linux]
+        - '*/parquet.dll'                # [win]
+        - '*/libgandiva.so.*'            # [linux]
+        - '*/gandiva.dll'                # [win]
```

Here we can actually specify the path (which is much shorter than for site-packages). I also think it'd be more readable to separate linux and win, and AFAICT we need to add osx:
```suggestion
        - 'lib/libarrow_acero.*'              # [unix]
        - 'lib/libarrow_dataset.*'            # [unix]
        - 'lib/libarrow_substrait.*'          # [unix]
        - 'lib/libarrow_flight.*'             # [unix]
        - 'lib/libparquet.*'                  # [unix]
        - 'lib/libgandiva.*'                  # [unix]
        - 'Library/lib/arrow_acero.dll'       # [win]
        - 'Library/lib/arrow_dataset.dll'     # [win]
        - 'Library/lib/arrow_substrait.dll'   # [win]
        - 'Library/lib/arrow_flight.dll'      # [win]
        - 'Library/lib/parquet.dll'           # [win]
        - 'Library/lib/gandiva.dll'           # [win]
```

---

Thanks! yeah, I missed macOS